### ISSUE TITLE
non-empty solution in `onNewSolution`

### ISSUE THREAD

**Issue by v4hn** [ros-planning/moveit_task_constructor#26]
After longer execution (2-3 minutes) of my 10+ stages serial container and possibly close to the end of exhaustive planning I currently see the following quite often:

```
moveit_task_constructor/core/src/container.cpp:347: virtual void moveit::task_constructor::SerialContainer::onNewSolution(const moveit::task_constructor::SolutionBase&): Assertion `solution.empty()' failed.
```

I did not investigate further yet, because I guess you have some ideas what the problem might be @rhaschke  .

**Comment by rhaschke** [comment]
For me the assertion never failed: As every solution should be moved into the sorted list, I cannot really see, why the assertion fails. However, I provided a quick fix in #27.
Maybe, you should provide your stage pipeline for further investigation.

**Comment by v4hn** [comment]
hopefully obsolete, I haven't seen the problem in a long time now.
